tests/unit/Environment/ServerTest.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Quantum\Tests\Environment;
+
+use Quantum\Environment\Server;
+use Quantum\Tests\AppTestCase;
+use ReflectionClass;
+
+class ServerTest extends AppTestCase
+{
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $reflection = new ReflectionClass(Server::class);
+        $instance = $reflection->getProperty('instance');
+        $instance->setAccessible(true);
+        $instance->setValue(null);
+    }
+
+    public function testGetInstance()
+    {
+        $server1 = Server::getInstance();
+        $server2 = Server::getInstance();
+
+        $this->assertSame($server1, $server2);
+    }
+
+    public function testAll()
+    {
+        $server = Server::getInstance();
+        $server->set('REQUEST_METHOD', 'GET');
+        $server->set('REQUEST_URI', '/test');
+
+        $allServerData = $server->all();
+
+        $this->assertArrayHasKey('REQUEST_METHOD', $allServerData);
+        $this->assertArrayHasKey('REQUEST_URI', $allServerData);
+
+        $this->assertEquals('GET', $allServerData['REQUEST_METHOD']);
+        $this->assertEquals('/test', $allServerData['REQUEST_URI']);
+    }
+
+    public function testGet()
+    {
+        $server = Server::getInstance();
+        $server->set('REQUEST_METHOD', 'POST');
+
+        $this->assertEquals('POST', $server->get('REQUEST_METHOD'));
+        $this->assertNull($server->get('NON_EXISTENT_KEY'));
+    }
+
+    public function testSet()
+    {
+        $server = Server::getInstance();
+        $server->set('CUSTOM_KEY', 'custom_value');
+
+        $this->assertEquals('custom_value', $server->get('CUSTOM_KEY'));
+    }
+
+    public function testUri()
+    {
+        $server = Server::getInstance();
+        $server->set('REQUEST_URI', '/test/uri');
+
+        $this->assertEquals('/test/uri', $server->uri());
+    }
+
+    public function testQuery()
+    {
+        $server = Server::getInstance();
+        $server->set('QUERY_STRING', 'foo=bar');
+
+        $this->assertEquals('foo=bar', $server->query());
+    }
+
+    public function testMethod()
+    {
+        $server = Server::getInstance();
+        $server->set('REQUEST_METHOD', 'PUT');
+
+        $this->assertEquals('PUT', $server->method());
+    }
+
```

Comment:
testMethod can be removed because testGet also tests method

Reply:
The `testGet` doe not test the `$server->method()`